From git-commits-head-owner@vger.kernel.org Mon Oct 23 10:18:42 2006
Date: Sun, 22 Oct 2006 00:59:04 GMT
Message-Id: <200610220059.k9M0x4e8025532@hera.kernel.org>
From: Linux Kernel Mailing List <linux-kernel@vger.kernel.org>
To: git-commits-head@vger.kernel.org
Subject: [PATCH] x86-64: Fix C3 timer test

commit aa026ede513b7d672fa7d9106b2f2a475455dcf2
tree 13a4e7934f3597336ebdfde1d720b5f2c877ad6e
parent cb7fabcf9d4340b61f012cc7e8e3d7ff5c4f441c
author Andi Kleen <ak@suse.de> 1161470475 +0200
committer Andi Kleen <andi@basil.nowhere.org> 1161470475 +0200

[PATCH] x86-64: Fix C3 timer test

There was a typo in the C3 latency test to decide of the TSC
should be used or not. It used the C2 latency threshold, not the
C3 one. Fix that.

This should fix the time on various dual core laptops.

Signed-off-by: Andi Kleen <ak@suse.de>

 arch/x86_64/kernel/time.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86_64/kernel/time.c b/arch/x86_64/kernel/time.c
index 1ba5a44..88722f1 100644
--- a/arch/x86_64/kernel/time.c
+++ b/arch/x86_64/kernel/time.c
@@ -948,7 +948,7 @@ #endif
  	if (boot_cpu_data.x86_vendor == X86_VENDOR_INTEL) {
 #ifdef CONFIG_ACPI
 		/* But TSC doesn't tick in C3 so don't use it there */
-		if (acpi_fadt.length > 0 && acpi_fadt.plvl3_lat < 100)
+		if (acpi_fadt.length > 0 && acpi_fadt.plvl3_lat < 1000)
 			return 1;
 #endif
  		return 0;

